krm-functions/lib/fn-runtime/upstream_inventory_test.go

```diff
@@ -0,0 +1,539 @@
+
+		ipa = &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: "type2",
+			},
+		}
+		byteStream, _ = yaml.Marshal(ipa)
+		kubeObjectMade, _ = fn.ParseKubeObject(byteStream)
+	}
+	diffList, _ := inventory.Diff()
+	if len(diffList.CreateObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToDelete: -want 0, +got:\n%v", len(diffList.CreateObjs))
+	}
+	if len(diffList.DeleteObjs) != 2 {
+		t.Errorf("TestDiffWithSpecToDelete: -want 2, +got:\n%v", len(diffList.DeleteObjs))
+	}
+	if len(diffList.UpdateObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToDelete: -want 0, +got:\n%v", len(diffList.UpdateObjs))
+	}
+}
+
+func TestDiffWithSpecToAdd(t *testing.T) {
+	inventory := NewUpstreamInventory()
+	type object struct {
+		apiVersion string
+		kind       string
+		name       string
+		dummy      string
+		spec       string
+	}
+
+	test1 := []struct {
+		input object
+	}{
+		{
+			input: object{
+				apiVersion: "a",
+				kind:       "b",
+				name:       "c",
+				dummy:      "sample",
+				spec:       "test1",
+			},
+		},
+		{
+			input: object{
+				apiVersion: "d",
+				kind:       "e",
+				name:       "f",
+				spec:       "test2",
+				dummy:      "sample",
+			},
+		},
+	}
+	for _, tt := range test1 {
+		currentGVKN := &corev1.ObjectReference{
+			APIVersion:      tt.input.apiVersion,
+			Kind:            tt.input.kind,
+			Name:            tt.input.name,
+			Namespace:       tt.input.dummy,
+			FieldPath:       tt.input.dummy,
+			ResourceVersion: tt.input.apiVersion,
+		}
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+
+		ipa = &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: "type2",
+			},
+		}
+		byteStream, _ = yaml.Marshal(ipa)
+		kubeObjectMade, _ = fn.ParseKubeObject(byteStream)
+		inventory.AddNewResource(currentGVKN, kubeObjectMade)
+	}
+	diffList, _ := inventory.Diff()
+	if len(diffList.CreateObjs) != 2 {
+		t.Errorf("TestDiffWithSpecToAdd: -want 2, +got:\n%v", len(diffList.CreateObjs))
+	}
+	if len(diffList.DeleteObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToAdd: -want 0, +got:\n%v", len(diffList.DeleteObjs))
+	}
+	if len(diffList.UpdateObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToAdd: -want 0, +got:\n%v", len(diffList.UpdateObjs))
```

Comment:
My understanding  as per code when there is new source and existing condition is nil,   the "CreateConditions" list needs to be populated as well. I may be off here though:)

Reply:
@ganchandrasekaran can you look at this?

Reply:
Merged the later into the same